### ISSUE TITLE
Docker information registered, but not adding External App.

### ISSUE THREAD

**Issue by corsairius** [nextcloud/llm2#11]
Docker information registered, but not adding External App.

I have the docker pull, resulting in this:
+-----+----------------------+---------------------+----------------+----------+----------------------+-------------------------------+
| Def | Name                 | Display name        | Deploy ID      | Protocol | Host                 | NC Url                        |
+-----+----------------------+---------------------+----------------+----------+----------------------+-------------------------------+
| *   | large_language_model | Docker Socket Proxy | docker-install | http     | /var/run/docker.sock | https://xyz.ddns.net |
+-----+----------------------+---------------------+----------------+----------+----------------------+-------------------------------+

Loaded the UI interface with the above and it is registered, but no External Apps are evident:

Deploy Daemon - Docker Socket Proxy
Default daemon. ExApps will be installed on it
ExApps installed: 0
Name: large_language_model
Protocol: http
Host: /var/run/docker.sock
Deploy config
Docker network: nextcloud
Nextcloud URL: https://xyz.ddns.net
HaProxy password: hsomepassword
GPUs support: false
Compute device: CPU

Disabled and Enabled AppAPI.
Connects but no External App.

Thanks!

**Comment by marcelklehr** [comment]
did you follow these steps? https://docs.nextcloud.com/server/latest/admin_manual/ai/app_llm2.html#installation

**Comment by corsairius** [comment]
0. Make sure the [Nextcloud Assistant app](https://docs.nextcloud.com/server/latest/admin_manual/ai/app_assistant.html#ai-app-assistant) is installed.  (This was already installed with llm v1)

1. [Install AppAPI and setup a Deploy Demon](https://docs.nextcloud.com/server/latest/admin_manual/ai/app_api_and_external_apps.html#ai-app-api) (Shows connectivity, but fails the Deploy Test with 
{"error":"ExApp not found, failed to get logs"}

3. Install the “Local large language model” ExApp via the “External Apps” page in the Nextcloud web admin user interface

All except for the last.  There is no External Apps application.

**Comment by bigcat88** [comment]
No, it is where it was from the very beginning..

<img width="396" alt="image" src="https://github.com/nextcloud/llm2/assets/13381981/658efe22-145d-4959-b99f-10468cfd68e6">

**Comment by corsairius** [comment]
I don't have that External Apps icon in the menu.

Forgot the nextcloud summary, Version [Nextcloud Hub 7](https://nextcloud.com/) (28.0.5)

**Comment by corsairius** [comment]
I believe I was missing this snippet "You will need to [setup a docker container called docker-socket-proxy that proxies access to docker for your Nextcloud instance](https://github.com/cloud-py-api/docker-socket-proxy#readme)."

`docker run -e NC_HAPROXY_PASSWORD="some_secure_password" \
  -v /var/run/docker.sock:/var/run/docker.sock \
  --name nextcloud-appapi-dsp -h nextcloud-appapi-dsp \
  --restart unless-stopped --privileged -d ghcr.io/cloud-py-api/nextcloud-appapi-dsp:release`

Still need to execute the proper Docker run command. This step isn't obvious, as it is recommended to use the UI instead of the CLI.  The occasional installer would likely miss this.  I followed what I thought were the instructions, and was not successful.  If the UI were to be used as indicated, it should take care of this.  

Now I know why my container from llm2 failed, duh...  Maybe.  I chose the wrong external link vs internal trusted network link since working in the library instead of home...  This external app stuff gives me the ability to use other GGUF files, too.  Great stuff when I get it working.  Maybe get more siblings to use what Nextcloud offers.  Thanks guys!
